packages/linter/src/generators/lint-project/lint-project.ts

```diff
@@ -51,7 +51,7 @@ export async function lintProjectGenerator(
   const projectConfig = readProjectConfiguration(tree, options.project);
 
   const lintFilePatterns = options.eslintFilePatterns;
-  if (isBuildableLibraryProject(projectConfig)) {
+  if (lintFilePatterns && isBuildableLibraryProject(projectConfig)) {
```

Comment:
This would break the existing functionality

Reply:
Sounds good, the `?? []` solution is fine by me.